nixos/modules/services/torrent/qbittorrent-nox.nix

```diff
@@ -0,0 +1,141 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.qbittorrent-nox;
+in {
+  options = {
+    services = {
+      qbittorrent-nox = {
+        enable = mkEnableOption (lib.mdDoc "qbittorrent-nox daemon");
+
+        web = {
+          port = mkOption {
+            type = types.port;
+            default = 8080;
+            description = lib.mdDoc ''
+              qbittorrent-nox web UI port.
+            '';
+          };
+          openFirewall = mkOption {
+            type = types.bool;
+            default = false;
+            description = lib.mdDoc ''
+              Whether to open the firewall for the ports in
+              {option}`services.qbittorrent-nox.web.port`.
+            '';
+          };
+        };
+
+        torrenting = {
+          port = mkOption {
+            type = types.port;
+            default = 48197;
+            description = lib.mdDoc ''
+              qbittorrent-nox web UI port.
+            '';
+          };
+
+          openFirewall = mkOption {
+            default = false;
+            type = types.bool;
+            description = lib.mdDoc ''
+              Whether to open the firewall for the ports in
+              {option}`services.qbittorrent-nox.torrenting.port`.
+            '';
+          };
+        };
+
+        dataDir = mkOption {
```

Comment:
is a dataDir true here for qbit, profiles seems to be the way it manages itself, a profilesdir commandline option is available.